deployments.go

```diff
@@ -58,25 +59,43 @@ func (s *deployerService) deploy(ctx context.Context, db *gorm.DB, opts DeployOp
 		Slug:        slug,
 		Description: desc,
 	})
-
 	return r, err
 }
 
+func (s *deployerService) deployInTransaction(ctx context.Context, stream scheduler.StatusStream, opts DeployOpts) (*Release, error) {
+	tx := s.db.Begin()
+	r, err := s.createRelease(ctx, tx, stream, opts)
+	if err != nil {
+		tx.Rollback()
+		return r, err
+	}
+	if err := tx.Commit().Error; err != nil {
+		return r, err
+	}
+	return r, s.releases.Release(ctx, r, stream)
+}
+
 // Deploy is a thin wrapper around deploy to that adds the error to the
 // jsonmessage stream.
-func (s *deployerService) Deploy(ctx context.Context, db *gorm.DB, opts DeployOpts) (*Release, error) {
+func (s *deployerService) Deploy(ctx context.Context, opts DeployOpts) (*Release, error) {
 	var msg jsonmessage.JSONMessage
 
-	r, err := s.deploy(ctx, db, opts)
+	var stream scheduler.StatusStream
+	if opts.Stream {
+		stream = scheduler.NewJSONMessageStream(opts.Output)
+	}
+
+	r, err := s.deployInTransaction(ctx, stream, opts)
 	if err != nil {
 		msg = newJSONMessageError(err)
-	} else {
+	} else if stream == nil {
 		msg = jsonmessage.JSONMessage{Status: fmt.Sprintf("Status: Created new release v%d for %s", r.Version, r.App.Name)}
```

Comment:
I feel like this message should always be present, regardless of whether streaming is enabled or not.